lib/spack/spack/filesystem_view.py

```diff
@@ -40,6 +41,44 @@
 _projections_path = '.spack/projections.yaml'
 
 
+def view_symlink(src, dst, view, spec):
+    os.symlink(src, dst)
+
+
+def view_hardlink(src, dst, view, spec):
+    os.link(src, dst)
+
+
+def view_copy(src, dst, view, spec):
+    shutil.copyfile(src, dst)
+    if spec:
+        # Not metadata, we have to relocate it
+
+        # What type of file are we relocating
+        relocate_method = spack.relocate.relocate_text_bin \
+            if spack.relocate.is_binary(dst) else spack.relocate.relocate_text
+
+        # Get information on where to relocate from/to
+        prefix_to_prefix = dict(
+            (dep.prefix, view.get_projection_for_spec(dep))
+            for dep in spec.traverse()
+        )
+
+        # Call actual relocation method
+        relocate_method(
+            [dst], spack.store.layout.root, view._root,
+            spec.prefix, view.get_projection_for_spec(spec),
+            spack.paths.spack_root, view._root, prefix_to_prefix
+        )
```

Comment:
Relocation might take some time. Since this function doesn't prompt anything up to completion, relocating a package might give users the feeling Spack is hanging:
```console
$ time spack view relocate ~/tmp/spack-relocate netlib-scalapack

real    0m25,616s
user    0m18,356s
sys     0m7,839s
```